mgmt2/rpc/jsonrpc/Defs.h

```diff
@@ -45,11 +45,20 @@ class RPCHandlerResponse
 };
 
 struct RPCResponseInfo {
-  RPCResponseInfo(std::optional<std::string> const &id_) : id(id_) {}
+  RPCResponseInfo(std::optional<std::string> const &id_) : id{id_} {}
+  RPCResponseInfo(std::error_code const &ec_) : error{ec_, {}} {}
+  RPCResponseInfo(std::optional<std::string> const &id_, std::error_code const &ec_) : error{ec_, {}}, id{id_} {}
+  RPCResponseInfo(std::optional<std::string> const &id_, std::error_code const &ec_, ts::Errata const &errata)
+    : error{ec_, errata}, id{id_}
+  {
+  }
   RPCResponseInfo() = default;
 
   RPCHandlerResponse callResult;
-  std::error_code rpcError;
+  struct Error {
+    std::error_code ec;
```

Comment:
We can fix this once we switch to `swoc::Errata`. It supports embedding the `ec` in the `Errata`.

Reply:
Yes, this is on my TODO list once we have libswoc in our private repo.